File: crates/sargon-uniffi/src/hierarchical_deterministic/cap26/cap26_path/paths/account_path.rs
use crate::prelude::*;
use sargon::AccountPath as InternalAccountPath;
use sargon::EntityCAP26Path;

#[derive(Clone, PartialEq, Eq, Hash, InternalConversion, uniffi::Record)]
pub struct AccountPath {
    pub path: HDPath,

    pub network_id: NetworkID,

    pub entity_kind: CAP26EntityKind,

    pub key_kind: CAP26KeyKind,

    pub index: HDPathValue,
}

#[uniffi::export]
pub fn new_account_path_sample() -> AccountPath {
    InternalAccountPath::sample().into()
}

#[uniffi::export]
pub fn new_account_path_sample_other() -> AccountPath {
    InternalAccountPath::sample_other().into()
}

#[uniffi::export]
pub fn new_account_path(
    network_id: NetworkID,
    key_kind: CAP26KeyKind,
    index: HDPathValue,
) -> AccountPath {
    InternalAccountPath::new(network_id.into(), key_kind.into(), index.into())
        .into()
}

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/AccountPath.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.AccountPath
import com.radixdlt.sargon.Cap26EntityKind
import com.radixdlt.sargon.Cap26KeyKind
import com.radixdlt.sargon.CommonException
import com.radixdlt.sargon.DerivationPath
import com.radixdlt.sargon.Hardened
import com.radixdlt.sargon.NetworkId
import com.radixdlt.sargon.newAccountPath

fun AccountPath.Companion.init(
    networkId: NetworkId,
    keyKind: Cap26KeyKind,
    index: Hardened
): AccountPath = newAccountPath(
    networkId = networkId,
    keyKind = keyKind,
    index = index
)

fun AccountPath.asGeneral(): DerivationPath = DerivationPath.Account(value = this)